tests/Unit/TestCase.php

```diff
@@ -19,24 +18,29 @@ abstract class TestCase extends BaseTestCase
      */
     protected function getEnvironmentSetUp($app)
     {
-        // Setup default database to use sqlite :memory:
-        $app['config']->set('database.default', 'testbench');
-        $app['config']->set('database.connections.testbench', [
-            'driver' => 'sqlite',
-            'database' => ':memory:',
-            'prefix' => '',
-        ]);
         $app['config']->set('logging.default', 'stderr');
     }
 
-    protected function setUp(): void
+    protected function getPackageProviders($app)
     {
-        parent::setUp();
-        $this->loadMigrationsFrom(__DIR__ . '../../../database/migrations');
+        return [ToolServiceProvider::class];
     }
 
-    protected function getPackageProviders($app)
+    protected function defineDatabaseMigrations()
     {
-        return [ToolServiceProvider::class];
+        // Since our migrations depends on the default laravel migrations, we
+        // cannot use the RefreshDatabase trait. Because that trait will wipe
+        // the default laravel mirgations.
+        // That is why we do a migrate and rollback here.
+        $this->loadLaravelMigrations([
+            '--database' => 'mysql',
+        ]);
+        $this->artisan('migrate', ['--database' => 'mysql'])->run();
+
+        $this->beforeApplicationDestroyed(function () {
+            $this->artisan('migrate:rollback', [
+                '--database' => 'mysql',
+            ])->run();
+        });
```

Comment:
Ik snap dit stuk niet helemaal. Want `loadLaravelMigrations()` lijkt al die migration en rollback uit te voeren.

Reply:
`loadLaravelMigrations()` voort alleen de migraties uit die worden meegeleverd met Laravel (users en password tables). Die zitten niet in onze package.

Daarna voeren we migrate artisan command uit. Die voert dan de migrations uit die in onze package zijn gedefinieerd.

Als laatste stap is de rollback van alle migraties.

Normaal wil je dat doen met de `RefreshDatabase` trait. Alleen het probleem met de `RefreshDatabase` trait is dat hij eerst alle tabellen wiped, inclusief de users and password tables (die worden gemaakt met `loadLaravelMigrations`).
Waardoor onze eigen migraties weer falen omdat we een foreign constraint proberen te zetten op de users tabel, die er dan niet meer is.

Dit was de grootste reden waarom ik hier zo mee aan het kutten was.

Reply:
Die comment geldt dus voor alle stappen in deze functie, wat misschien een beetje onduidelijk is. Ik zie nu ook een spelfout in de comment